runtime/float.go

```diff
@@ -355,6 +374,39 @@ func floatDivModOp(f *Frame, method string, v, w *Object, fun func(v, w float64)
 	return NewFloat(x).ToObject(), nil
 }
 
+func hashFloat(v float64) int {
+	if math.IsNaN(v) {
+		return 0
+	}
+
+	if math.IsInf(v, 0) {
+		if math.IsInf(v, 1) {
+			return 314159
+		} else if math.IsInf(v, -1) {
```

Comment:
When one branch returns, prefer to elide the else to minimize indentation: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

```
if math.IsInf(v, 1) {
  return 314159
}
if math.IsInf(v, -1) {
  return -271828
}
return 0
```